File: examples/hello.R
# R example - Statistical computing and graphics!

cat("📊 Hello from R!\n")
cat("R: Language for data analysis and statistics\n\n")

# Variables
message <- "Welcome to AI-IDE"
cat(message, "\n")

# Function
add <- function(a, b) {
  return(a + b)
}

sum_result <- add(5, 10)
cat("5 + 10 =", sum_result, "\n")

# Vector example
numbers <- c(1, 2, 3, 4, 5)
total <- sum(numbers)
cat("Sum of vector:", total, "\n")

# Loop example
for (num in numbers) {
  cat("Number:", num, "\n")
}

# Data frame example
df <- data.frame(
  name = c("Alice", "Bob", "Charlie"),
  age = c(25, 30, 35),
  score = c(85, 90, 95)
)

cat("\nData frame:\n")
print(df)

# Statistical summary
cat("\nSummary:\n")
print(summary(df$age))

File: examples/hello.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

int main() {
    std::cout << "⚡ Hello from C++!" << std::endl;
    std::cout << "==============================" << std::endl;
    
    // Variables
    std::string projectName = "AI-IDE";
    std::cout << "Welcome to " << projectName << "!" << std::endl;
    
    // Vector operations
    std::vector<int> numbers = {1, 2, 3, 4, 5};
    std::vector<int> squares;
    
    // Transform to squares
    std::transform(numbers.begin(), numbers.end(), 
                   std::back_inserter(squares),
                   [](int n) { return n * n; });
    
    std::cout << "Numbers: ";
    for (size_t i = 0; i < numbers.size(); ++i) {
        std::cout << numbers[i];
        if (i < numbers.size() - 1) std::cout << ", ";
    }
    std::cout << std::endl;
    
    std::cout << "Squares: ";
    for (size_t i = 0; i < squares.size(); ++i) {
        std::cout << squares[i];
        if (i < squares.size() - 1) std::cout << ", ";
    }
    std::cout << std::endl;
    
    // String operations
    std::string message = "C++ execution in AI-IDE works great!";
    std::cout << "Message: " << message << std::endl;
    std::cout << "Length: " << message.length() << std::endl;
    
    std::cout << "\n✅ C++ execution complete!" << std::endl;
    return 0;
}